src/core/thread/mle_router_ftd.hpp

```diff
@@ -65,6 +65,18 @@ class NetworkDataLeader;
  */
 
 /**
+* This structure represents the child information for persistent storage.
+*
+*/
+struct ChildInfo
+{
+    Mac::ExtAddress  mExtAddress;    ///< Extended Address
+    uint32_t         mTimeout;       ///< Timeout
+    uint16_t         mRloc16;        ///< RLOC16
+    uint8_t          mMode : 4;      ///< The MLE device mode
```

Comment:
You probably don't want to update the Neighbor class.  The bits are packed to minimize memory usage.  The settings field is a bit different because that structure definition needs to span across different firmware versions and the behavior of appending new fields to existing bit-fields is not terribly obvious.